src/main/java/org/prebid/server/bidder/telaria/TelariaBidder.java

```diff
@@ -72,24 +74,18 @@ public TelariaBidder(String endpointUrl, JacksonMapper mapper) {
         for (Imp imp : bidRequest.getImp()) {
             try {
                 final ExtImpTelaria extImp = parseImpExt(imp);
-                if (StringUtils.isBlank(extImp.getSeatCode())) {
-                    throw new PreBidException("Telaria: Seat Code required");
-                }
                 seatCode = extImp.getSeatCode();
-                requestBuilder
-                        .imp(Collections.singletonList(imp.toBuilder()
-                                .tagid(extImp.getAdCode())
-                                .ext(mapper.mapper().valueToTree(ExtImpOutTelaria.of(imp.getTagid(), publisherId)))
-                                .build()));
+                Imp updateImp = updateImp(imp, extImp, publisherId);
+                requestBuilder.imp(Collections.singletonList(updateImp));
```

Comment:
You are still overriding imps. 
Just extract this method and return Collection of imps and then set it in the request